Client/Frontend/Browser/TabManager.swift

```diff
@@ -829,21 +588,42 @@ extension TabManager {
                 }
             }
         }
+        guard count == 0
+            && !AppConstants.IsRunningTest
+            && !DebugSettingsBundleOptions.skipSessionRestore else {
+                return
+        }
+        // This is wrapped in an Objective-C @try/@catch handler because NSKeyedUnarchiver may throw exceptions which Swift cannot handle
+        _ = Try(
+            withTry: { () -> Void in
+                guard let savedTabs = self.store.tabsToRestore(
+                    fromData: self.store.tabArchiveData()) else {
+                        return
+                }
+                self.restoreInternal(
+                    savedTabs: savedTabs, clearPrivateTabs: self.shouldClearPrivateTabs())
+            },
+            catch: { exception in
+                Sentry.shared.send(message: "Failed to restore tabs: ", tag: SentryTag.tabManager, severity: .error, description: "\(exception ??? "nil")")
+            }
+        )
+    }
 
-        guard let savedTabs = TabManager.tabsToRestore() else {
+    fileprivate func restoreInternal(savedTabs: [SavedTab], clearPrivateTabs: Bool) {
+        guard !savedTabs.isEmpty else {
             return
         }
 
-        if count == 0 && !AppConstants.IsRunningTest && !DebugSettingsBundleOptions.skipSessionRestore {
-            // This is wrapped in an Objective-C @try/@catch handler because NSKeyedUnarchiver may throw exceptions which Swift cannot handle
-            _ = Try(
-                withTry: { () -> Void in
-                    self.restoreInternal(savedTabs: savedTabs, clearPrivateTabs: self.shouldClearPrivateTabs())
-                },
-                catch: { exception in
-                    Sentry.shared.send(message: "Failed to restore tabs: ", tag: SentryTag.tabManager, severity: .error, description: "\(exception ??? "nil")")
-                }
-            )
+        let tabToSelect = store.restoreInternal(
+            savedTabs: savedTabs, clearPrivateTabs: clearPrivateTabs, tabManager: self)
```

Comment:
Im not sure we use this kind of style anywhere. To be more consistent we should have it on the same line. (I think there are a few places where this happens)